controller/src/main/java/io/pravega/controller/eventProcessor/impl/EventProcessorCell.java

```diff
@@ -230,9 +232,12 @@ void store(Position position) {
                        final String readerId,
                        final int index,
                        final CheckpointStore checkpointStore) {
-        this.reader = reader;
-        this.selfWriter = selfWriter;
-        this.checkpointStore = checkpointStore;
+        Preconditions.checkNotNull(eventProcessorConfig);
```

Comment:
IMO the actual class should implement its correctness/validation logic. All the business logic specific to the class should be self-contained.